### PR TITLE
#1 Switch to Redux toolkit and Typescript 3.8

### PR DESCRIPTION
**Pull request by deonclem**

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -5,26 +5,16 @@ import * as serviceWorker from "./serviceWorker"
...
-
-const store = createStore(rootReducer, applyMiddleware(sagaMiddleware))
-
-sagaMiddleware.run(rootSaga)
-
 const I18nWrapper: React.FC = () => {
-  const { locale } = useSelector<any, LocaleReducerState>(
-    store => store.localeReducer,
-  )
+  const { locale } = useSelector<AppState, LocaleState>(store => store.locale)
```
Comment (markerikson) [inline src/index.tsx:17]: There's a few issues with this selector usage:

- Semantically, the argument isn't the `store`.  It's the `state` that's inside that store.  Strongly recommend changing the variable name to `state` instead.
- You generally shouldn't manually declare the generics up front.  Instead, declare it as `const someValue = useSelector( (state: AppState) => state.slice.someValue)`.  The return type will be inferred automatically.
- By returning a larger object and then destructuring the result, the component will re-render whenever that larger value is updated.  This is not usually what you want - you should select the smallest possible piece of state this component depends on, like: `const locale = useSelector( (state: AppState) => state.locale.locale)`.

File: src/reducers/localeSlice.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit"
+
+export interface LocaleState {
+  locale: string
```
Comment (markerikson) [inline src/reducers/localeSlice.ts:4]: Do you actually plan on storing more values besides this one string?  If not, the wrapping object isn't necessary - the entire slice state could just be the one string.

File: src/store.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+import rootReducer from "./reducers/rootReducer"
+import createSagaMiddleware from "redux-saga"
+import rootSaga from "./sagas/rootSaga"
+
+const sagaMiddleware = createSagaMiddleware()
+
+const store = configureStore({
+  reducer: rootReducer,
+  middleware: [sagaMiddleware],
```
Comment (markerikson) [inline src/store.ts:11]: Note that this is leaving out any of the default middleware that RTK comes with, including both the thunk middleware and the mutation checking middleware.  I see that you're using sagas, so thunks may not be as critical, but it's worth keeping the others in there.  To do that, import `getDefaultMiddleware`, and change this line to:

```ts
  middleware: [...getDefaultMiddleware({thunk: false}), sagaMiddleware]
```

File: src/index.tsx (hunk 1 above)
Comment (deonclem) [reply src/index.tsx:17]: Fixed all remarks.

File: src/reducers/localeSlice.ts (hunk 2 above)
Comment (deonclem) [reply src/reducers/localeSlice.ts:4]: Not necessarily you're right - I changed it to a string.

**Pull request merged**